cabal.project

```diff
@@ -15,7 +15,6 @@ packages: semantic
           semantic-ruby
           semantic-rust
           semantic-scope-graph
-          semantic-source
```

Comment:
Here, and in `cabal.project.ci`, we remove `semantic-source` from the top-level project, ensuring that it'll be fetched from hackage. This makes it possible to factor out PRs like this one, at the cost of making it less convenient to change things across both `semantic` (or any other local package) and `semantic-source`.

For this PR I think that's worth it; for the next one I might end up reinstating this. But I wanted to factor the PRs out to begin with to limit scope and to be sure we're handling versioning appropriately.

Reply:
> at the cost of making it less convenient to change things across both

To clarify, this means that the process would be to merge the PR that touches the "downstream" package, cut a new release and deploy it to hackage, and _then_ open the PR in the "upstream" package?

(Not meaning to suggest a different approach, just making sure I understand how the process would be impacted.)